resources/[system]/scoreboard/client.lua

```diff
@@ -0,0 +1,131 @@
+local isShowing = false
+local players = {}
+local lastUpdate = 0
+
+RegisterKeyMapping("+scoreboard", "Open the scoreboard.", "keyboard", "z")
+RegisterCommand("+scoreboard", function(source, args, rawcommand)
+  if not isShowing then
+    if GetGameTimer() >= lastUpdate + 3000 then
+      lastUpdate = GetGameTimer()
+      TriggerServerEvent("scoreboard:getPlayers")
+    end
+    sortedColumns = sortColumns(GlobalState.columns)
+    SendNUIMessage({
+      app = 'CfxScoreboard',
+      method = 'setColumns',
+      data = sortedColumns
+    })
+    SendNUIMessage({
+      app = 'CfxScoreboard',
+      method = 'setVisibility',
+      data = true
+    })
+    SetNuiFocus(true, true)
+    SetNuiFocusKeepInput(true)
+    SetCursorLocation(0.5, 0.5)
+    isShowing = true
+    Citizen.CreateThread(function()
+      while isShowing do
+        Citizen.Wait(0)
+        DisableControlAction(0, 1, true) -- INPUT_LOOK_LR - Prevent moving camera when moving mouse.
+        DisableControlAction(0, 2, true) -- INPUT_LOOK_UD - Prevent moving camera when moving mouse.
+        DisableControlAction(0, 16, true) -- INPUT_SELECT_NEXT_WEAPON - Prevent switching weapon when scrolling through list.
+        DisableControlAction(0, 17, true) -- INPUT_SELECT_PREV_WEAPON	 - Prevent switching weapon when scrolling through list.
```

Comment:
It may be a good idea to disable chat from being opened so they don't overlap. I'm not sure if any issues would arise in terms of the mouse losing focus when it's closed as I haven't tested this, but it's probably worth doing.

DisableControlAction(0, 255, true) -- INPUT_MP_TEXT_CHAT_ALL    - Prevent chat from opening.

And maybe the pause menu as well if closing of the scoreboard is ever handled by `ESCAPE` but I didn't see any code pertaining to it so it would be fine to not do it.

There's also the switching of weapons while in a vehicle too as players won't always be on foot:
DisableControlAction(0, 99, true) -- INPUT_VEH_SELECT_NEXT_WEAPON - Prevent switching weapon when scrolling through list.
DisableControlAction(0, 115, true) -- INPUT_VEH_FLY_SELECT_NEXT_WEAPON - Prevent switching weapon when scrolling through list.

I'm not sure if players will ever highlight anything on the scoreboard but it may be a good idea to also disable attacking while it's open.